File: java-version/src/main/java/com/example/java_version/controller/BasicAuthController.java
package com.example.java_version.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/basic")
public class BasicAuthController {

    @PostMapping("/login")
    public ResponseEntity<Void> login() {
        return ResponseEntity.ok().build();
    }
}

File: kotlin-version/src/main/kotlin/com/example/kotlin_version/controller/BasicAuthController.kt
package com.example.kotlin_version.controller

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/basic")
class BasicAuthController {

    @PostMapping("/login")
    fun login(): ResponseEntity<Void> {
        return ResponseEntity.ok().build()
    }
}